<launch>
	<!--<arg name="map" default="$(find searcher)/maps/house_map.yaml"/>-->

	<node name="reset_shm" pkg="velma_common" type="reset_shm_comm"/>

	<node name="init" pkg="velma_task_cs_ros_interface" type="initialize_robot.py"/>

	<include file="$(find searcher)/launch/velma_system_mobile.launch" />

	<!--<node name="map_server" pkg="map_server" type="map_server" args="$(arg map)" />-->

	<param name="/sim_mode" type="string" value="gazebo" />
	

	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find searcher)/rviz/rviz.rviz"/>

	<!-- manipulative part of the system -->

	<!-- octomap server -->
	<include file="$(find velma_common)/launch/octomap_server.launch" />

	<!-- trajectory planner -->
	<include file="$(find velma_ros_plugin)/launch/velma_planner.launch" />


	<!-- mobile part of the system -->

	<node name="map_odom_tf_publisher" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 map odom 10" />

	<!-- static world to torso tf publisher-->
	<node name="world_torso_tf_publisher" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 world torso_base 100" />

	<arg name="nodelet_manager_name" value="nodelet_manager"/>
	<arg name="publish_nav_vel_topic" value="/mux_vel_nav/cmd_vel"/>
 	
	<node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager" />
	 <node pkg="nodelet" type="nodelet" name="yocs_cmd_vel_mux" args="load yocs_cmd_vel_mux/CmdVelMuxNodelet $(arg nodelet_manager_name)" output="screen">
    	<param name="yaml_cfg_file" value="$(find searcher)/param/cmd_mux_config.yaml"/>
		<!--<remap from="/yocs_cmd_vel_mux/mux_vel_keyboard/cmd_vel" to="/mux_vel_keyboard/cmd_vel"/>
		<remap from="/yocs_cmd_vel_mux/mux_vel_nav/cmd_vel" to="$(arg publish_nav_vel_topic)"/>
		<remap from="/yocs_cmd_vel_mux/mux_vel_raw/cmd_vel" to="/mux_vel_raw/cmd_vel"/>
		<remap from="/cmd_vel_absolute" to="/cmd_vel"/>-->
	</node>

</launch>
